### ISSUE TITLE
Tabulator custom formatter throws "Uncaught TypeError: setting getter-only property" when data class has getter property

### ISSUE THREAD

**Issue by cuchaz** [rjaros/kvision#275]
Hello! =)

Here's a minimum working example:
```kotlin
data class Foo(val bar: String) {
	val baz: String get() = "cheese"
}
val data = listOf(
	Foo("cow"),
	Foo("moo")
)
tabulator(
	data,
	options = TabulatorOptions(
		columns = listOf(
			ColumnDefinition(
				"Foo",
				formatterComponentFunction = { _, _, foo ->
					Span(foo.toString())
				}
			)
		)
	)
)
```
If you remove the `baz` getter property, everything works fine. If you remove the `formatterComponentFunction`, everything works fine too.
Here's the relevant bits of the javascript stack trace from the browser:
```
Uncaught TypeError: setting getter-only property "baz"
    toKotlinObj          webpack://project/./kotlin-dce-dev/kvision-js-legacy.js?:24985
    lambda_0             webpack://project/./kotlin-dce-dev/kvision-kvision-tabulator-js-legacy.js?:2202
    formatValue          webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:14
    _generateContents    webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:9
    layoutElement        webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:10
    getElement           webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:9
    initialize           webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:9
    initialize           webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:9
    _virtualRenderFill   webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:8
    reRenderInPosition   webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:8
    redraw               webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:9
    redraw               webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:11
    containerObserver    webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:17
    initialize           webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:17
    _buildElement        webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:10
    _create              webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:10
    e                    webpack://project/../../node_modules/tabulator-tables/dist/js/tabulator.min.js?:10
    createInstance       webpack://project/./kotlin-dce-dev/kvision-js-legacy.js?:24938
    createJsTabulator    webpack://project/./kotlin-dce-dev/kvision-kvision-tabulator-js-legacy.js?:4358
    afterInsert_cmtal8$  webpack://project/./kotlin-dce-dev/kvision-kvision-tabulator-js-legacy.js?:4382
    lambda_0             webpack://project/./kotlin-dce-dev/kvision-js-legacy.js?:8039
    patch                webpack://project/../../node_modules/snabbdom/es/snabbdom.js?:311
    patch_aevwwv$        webpack://project/./kotlin-dce-dev/kvision-js-legacy.js?:541
    reRender_8be2vx$     webpack://project/./kotlin-dce-dev/kvision-js-legacy.js?:21729
    refresh              webpack://project/./kotlin-dce-dev/kvision-js-legacy.js?:8484
    addInternal_87eobz$  webpack://project/./kotlin-dce-dev/kvision-js-legacy.js?:21910
    add_87eobz$          webpack://project/./kotlin-dce-dev/kvision-js-legacy.js?:21922
```

**Comment by rjaros** [comment]
Generally Tabulator support for KVision components (with `formatterComponentFunction` or `editorComponentFunctions` parameters) works only with "plain" data classes (without additional properties). It's because some "hacks" are used to convert Kotlin data classes to plain JS objects. But I'll take a look at your case - perhaps it could be easily fixed.

**Comment by cuchaz** [comment]
Cool, thanks!

But if it's too much trouble, don't worry about it. I've already implemented a workaround for my project, so it's not like I'm waiting for a solution.

Basically, I give a "plain" data class to the Tabulator that just wraps an Int index into the list of my real data classes. Then I resolve the index to the real data class inside the `formatterComponentFunction`.

It seems like there should be a simple way to filter out getter properties in JS-land though, so maybe it's easy to fix. =)

**Comment by joerg-rade** [comment]
Last week I stumbled across the same issue and used a similar workaround ;-)

**Comment by rjaros** [comment]
This should be fixed in 4.8.3, because it's connected to #278
